app/models/docker_builder_service.rb

```diff
@@ -60,6 +61,9 @@ def push_image(tag)
         build.docker_repo_digest = "#{project.docker_repo}@#{matches[1]}"
       end
     end
+    # Fallbacks if registry did not send us back a digest
+    build.docker_repo_digest ||= result.info['RepoDigests'].first if result.info['RepoDigests']
+    build.docker_repo_digest ||= result.info['RepoTags'].first if result.info['RepoTags']
```

Comment:
Also, the `docker_ref` column is intended to refer to the tag of the image.

:-1: